cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1369,13 +1383,23 @@ void reader::impl::generate_list_column_row_count_estimates()
   // gives us the absolute row index
   // Note: chunk_row is already computed if we have column indexes
   if (not _has_page_index) {
+    thrust::for_each(rmm::exec_policy(_stream),
+                     pass.pages.d_begin(),
+                     pass.pages.d_end(),
+                     set_list_row_count_estimate{pass.chunks});
     auto key_input  = thrust::make_transform_iterator(pass.pages.d_begin(), get_page_chunk_idx{});
     auto page_input = thrust::make_transform_iterator(pass.pages.d_begin(), get_page_num_rows{});
     thrust::exclusive_scan_by_key(rmm::exec_policy_nosync(_stream),
                                   key_input,
                                   key_input + pass.pages.size(),
                                   page_input,
                                   chunk_row_output_iter{pass.pages.device_ptr()});
+  } else {
+    // If column indexes are available, we can translate PageInfo::chunk_row to PageInfo::num_rows
+    thrust::for_each(rmm::exec_policy_nosync(_stream),
+                     thrust::counting_iterator<size_t>(0),
+                     thrust::counting_iterator(pass.pages.size()),
+                     compute_page_num_rows_from_chunk_rows{pass.pages, pass.chunks});
```

Comment:
I vaguely don't like the idea of putting an accurate computation for row counts inside an function named `estimates`.  This is probably the right place for it though, so maybe the function name should be more specific.

Reply:
Updated the name of the function with an input argument indicating if we are writing estimates or not

Reply:
Oh.  I was referring to `generate_list_column_row_count_estimates()`